### PR TITLE
feat: expose requirepartitionfilter field in table

### PR DESCRIPTION
**Pull request by pmakani**

### CODE REVIEW

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableInfo.java [hunk 1]
```diff
@@ -305,6 +317,7 @@ public TableInfo build() {
     this.definition = builder.definition;
     this.encryptionConfiguration = builder.encryptionConfiguration;
     labels = builder.labels;
+    this.requirePartitionFilter = builder.requirePartitionFilter;
   }
```
Comment (dmitry-fa) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableInfo.java:321]: nit: switch two last lines:
```
    this.requirePartitionFilter = builder.requirePartitionFilter;
    labels = builder.labels;
```

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableInfo.java [hunk 2]
```diff
@@ -399,6 +412,14 @@ public BigInteger getNumRows() {
     return labels.userMap();
   }
 
+  /**
+   * If set to true, queries over this table require a partition filter (that can be used for
+   * partition elimination) to be specified.
+   */
+  public Boolean getRequirePartitionFilter() {
+    return requirePartitionFilter;
```
Comment (dmitry-fa) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableInfo.java:420]: The meet the javadoc style guide the comment should start with a verb.
I would rephrase:
```
/**
  * Returns true if a partition filter (that can be used for partition elimination) is required 
  * to be specified for queries over this table.
  */
```

**Review by dmitry-fa — CHANGES_REQUESTED** [review]

**Review by rahulKQL — APPROVED** [review]

**Comment by rahulKQL** [comment]
Considering you would take care of @dmitry-fa, rest LGTM

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableInfo.java (hunk 2 above)
Comment (pmakani) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableInfo.java:420]: it's copied from TimePartitioning

Comment (pmakani) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableInfo.java:420]: Done Thanks
